LoRaEngine/LoraKeysManagerFacade/CreateEdgeDevice.cs

```diff
@@ -4,47 +4,33 @@
 namespace CreateDeviceFunction
 {
     using System;
-    using System.Collections.Generic;
-    using System.IO;
     using System.Net;
     using System.Net.Http;
     using System.Text;
+    using System.Threading.Tasks;
     using Microsoft.AspNetCore.Http;
-    using Microsoft.AspNetCore.Mvc;
     using Microsoft.Azure.Devices;
     using Microsoft.Azure.Devices.Shared;
     using Microsoft.Azure.WebJobs;
     using Microsoft.Azure.WebJobs.Extensions.Http;
-    using Microsoft.Azure.WebJobs.Host;
-    using Microsoft.Extensions.Configuration;
+    using Microsoft.Build.Framework;
     using Newtonsoft.Json;
 
     public static class CreateEdgeDevice
     {
         [FunctionName("CreateEdgeDevice")]
-        public static async System.Threading.Tasks.Task<HttpResponseMessage> RunAsync([HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)]HttpRequest req, TraceWriter log, ExecutionContext context)
+        public static async Task<HttpResponseMessage> CreateEdgeDeviceImp([HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)]HttpRequest req, ILogger log, ExecutionContext context)
         {
-            var config = new ConfigurationBuilder()
-                .SetBasePath(context.FunctionAppDirectory)
-                .AddJsonFile("local.settings.json", optional: true, reloadOnChange: true)
-                .AddEnvironmentVariables()
-                .Build();
-            string connectionString = config.GetConnectionString("IoTHubConnectionString");
-            string deviceConfigurationUrl = Environment.GetEnvironmentVariable("DEVICE_CONFIG_LOCATION");
-            RegistryManager manager = RegistryManager.CreateFromConnectionString(connectionString);
+            var manager = LoraKeysManagerFacade.LoRaRegistryManager.GetCurrentInstance(context.FunctionAppDirectory);
+
             // parse query parameter
             var queryStrings = req.GetQueryParameterDictionary();
-            string deviceName = string.Empty;
-            string publishingUserName = string.Empty;
-            string publishingPassword = string.Empty;
-            string region = string.Empty;
-            string resetPin = string.Empty;
-
-            queryStrings.TryGetValue("deviceName", out deviceName);
-            queryStrings.TryGetValue("publishingUserName", out publishingUserName);
-            queryStrings.TryGetValue("publishingPassword", out publishingPassword);
-            queryStrings.TryGetValue("region", out region);
-            queryStrings.TryGetValue("resetPin", out resetPin);
+
+            queryStrings.TryGetValue("deviceName", out string deviceName);

```

Comment:
nit: any reason to use `string` instead of `var`?

Reply:
here I think it's hard to see what the type will be if you do var. To me this is easier to read and string is short enough.